nautobot/extras/plugins/__init__.py

```diff
@@ -623,3 +633,28 @@ def register_custom_validators(class_list):
             raise TypeError(f"PluginCustomValidator class {custom_validator} does not define a valid model!")
 
         registry["plugin_custom_validators"][custom_validator.model].append(custom_validator)
+
+
+def register_override_views(override_views, plugin):
+    resolver = get_resolver()
+
+    for qualified_view_name, view in override_views.items():
+        try:
+            app_name, view_name = qualified_view_name.rsplit(":", 1)
+        except ValueError:
+            raise ValidationError(
+                f"Plugin {plugin} tried to override view {qualified_view_name} "
+                f"but only top level namespace views are supported (e.g. `dcim:device`)."
+            )
+        while ":" in app_name:
+            resolver_name, app_name = app_name.split(":", 1)
+            resolver = resolver.namespace_dict[resolver_name][1]
+        app_resolver = resolver.namespace_dict.get(app_name)
+        if not app_resolver:
+            raise ValidationError(
```

Comment:
I wonder if we should make this just log an error message instead of aborting? I'm thinking of a forwards/backwards compatibility requirement, where we might want to potentially override a view that may or may not actually exist...